cli/src/cmd/forge/script/broadcast.rs

```diff
@@ -0,0 +1,260 @@
+    }
+
+    pub async fn handle_broadcastable_transactions(
+        &self,
+        target: &ArtifactId,
+        result: ScriptResult,
+        decoder: &mut CallTraceDecoder,
+        script_config: &ScriptConfig,
+    ) -> eyre::Result<()> {
+        if let Some(txs) = result.transactions {
+            println!("==========================");
+            println!("Simulated On-chain Traces:\n");
+            if let Ok(gas_filled_txs) = self.execute_transactions(txs, script_config, decoder).await
+            {
+                println!("\n\n==========================");
+                if !result.success {
+                    eyre::bail!("\nSIMULATION FAILED");
+                } else {
+                    let txs = gas_filled_txs;
+                    let mut deployment_sequence =
+                        ScriptSequence::new(txs, &self.sig, target, &script_config.config.out)?;
+                    deployment_sequence.save()?;
+
+                    if self.broadcast {
+                        self.send_transactions(&mut deployment_sequence).await?;
+                    } else {
+                        println!("\nSIMULATION COMPLETE. To broadcast these transactions, add --broadcast and wallet configuration(s) to the previous command. See forge script --help for more.");
+                    }
+                }
+            } else {
+                eyre::bail!("One or more transactions failed when simulating the on-chain version. Check the trace by re-running with `-vvv`")
+            }
+        } else if self.broadcast {
+            eyre::bail!("No onchain transactions generated in script");
+        }
+        Ok(())
+    }
+}
+
+pub fn set_chain_id(tx: &mut TypedTransaction, chain_id: u64) -> eyre::Result<()> {
+    match tx {
+        TypedTransaction::Legacy(tx) => tx.chain_id = Some(chain_id.into()),
+        TypedTransaction::Eip1559(tx) => tx.chain_id = Some(chain_id.into()),
+        _ => eyre::bail!("Wrong transaction type for expected output"),
+    }
+    Ok(())
+}
+
+pub fn into_legacy(tx: TypedTransaction) -> eyre::Result<TransactionRequest> {
+    Ok(match tx {
+        TypedTransaction::Legacy(tx) => tx,
+        _ => eyre::bail!("Wrong transaction type for expected output"),
+    })
+}
+
+pub fn into_legacy_ref(tx: &TypedTransaction) -> eyre::Result<&TransactionRequest> {
+    Ok(match tx {
+        TypedTransaction::Legacy(ref tx) => tx,
+        _ => eyre::bail!("Wrong transaction type for expected output"),
+    })
+}
+
+pub fn into_1559(tx: TypedTransaction) -> eyre::Result<Eip1559TransactionRequest> {
+    Ok(match tx {
+        TypedTransaction::Legacy(tx) => Eip1559TransactionRequest {
+            from: tx.from,
+            to: tx.to,
+            value: tx.value,
+            data: tx.data,
+            nonce: tx.nonce,
+            ..Default::default()
+        },
+        TypedTransaction::Eip1559(tx) => tx,
+        _ => eyre::bail!("Wrong transaction type for expected output"),
+    })
+}
```

Comment:
Should we add `as_1559`, `into_1559`, `as_legacy`, `into_legacy` helpers in ethers-rs? similar to the From impls? https://github.com/gakonst/ethers-rs/blob/master/ethers-core/src/types/transaction/eip2718.rs#L361-L377

Reply:
yes that would be ideal